### PR TITLE
Upgrade PixiJS, upgrade Material, lock down all package versions

### LINKED ISSUE #184: Update PIXI scaling mode from PIXI.SCALE_MODES.DEFAULT to PIXI.settings.SCALE_MODE
pixi deprecated the old method, but this also is currently broken in pixi 4.2.2. We'll have to wait until it's been resolved.

### LINKED ISSUE #247: Asset service does not load in parallel anymore
The resource loader library being used can load in parallel but you must not call load() until all the resources you want to load are added with add(). Currently we don't do this and simply call load after each resource is added. To fix this we will have to rework the asset service internals a bit.

### PR DESCRIPTION
**Pull request by tlein**
Resolves #184 
Resolves #247

### CODE REVIEW

File: src/duckling/project/asset.service.ts [hunk 1]
```diff
@@ -157,6 +164,15 @@ export class AssetService {
         return this._assets;
     }
 
+    private _onLoadedComplete(asset : Asset, editorSpecific : boolean) {
+        this._onAssetLoaded(asset, editorSpecific);
+        if (this._assetsToLoad.length > 0) {
```
Comment (jeffswenson) [inline src/duckling/project/asset.service.ts]: You should queue all unloaded assets at the same time.

File: src/index.html [hunk 2]
```diff
@@ -21,7 +21,15 @@
...
       md-dialog-container {
-          height: auto !important;
+        height: auto !important;
+      }
+      
+      /**
+       * Workaround for bug with how angular2-material is handling
+       * a non floating label for inputs
+       */
+      .mat-input-placeholder-wrapper::after {
```
Comment (jeffswenson) [inline src/index.html]: You should remove this hack. Let's disable the inline input label.

Comment (tlein) [reply src/index.html]: fixed

File: src/duckling/project/asset.service.ts (hunk 1 above)
Comment (tlein) [reply src/duckling/project/asset.service.ts]: fixed

File: src/duckling/project/asset.service.ts [hunk 3]
```diff
@@ -40,39 +47,46 @@ export class AssetService {
...
+                if (assetToLoad.asset.type === "FontTTF") {
+                    this._loadFont(assetToLoad);
                 } else {
-                    this._assetsToLoad.push({asset, filePath, editorSpecific});
+                    loader.add(assetToLoad.asset.key, assetToLoad.filePath);
                 }
             }
         }
+        
+        if (assets.length > 0) {
```
Comment (jeffswenson) [inline src/duckling/project/asset.service.ts]: This starts the loader even if the only asset loaded is a font.

File: src/duckling/project/asset.service.ts [hunk 4]
```diff
@@ -40,39 +47,46 @@ export class AssetService {
...
      * @param  editorSpecific Optional boolean that says if the resource is an editor specific resource, default is false.
      */
-    add(asset : Asset, filePath? : string, editorSpecific? : boolean) {
-        filePath = filePath || this._store.getState().project.home + "/resources/" + asset.key + "." + this._fileExtensionFromType(asset.type);
-        if (editorSpecific) {
-            asset.key = EDITOR_SPECIFIC_IMAGE_PREFIX + asset.key;
+    add(assets : LoadingAsset[]) {
+        if (loader.loading) {
+            for (let assetToLoad of assets) {
+                if (!this._assetsToLoad[assetToLoad.asset.key]) {
```
Comment (jeffswenson) [inline src/duckling/project/asset.service.ts]: This if is unnecessary.

File: src/duckling/project/asset.service.ts [hunk 5]
```diff
@@ -40,39 +47,46 @@ export class AssetService {
      * @param  filePath Optional filepath the asset is located. The default is /resources/<asset_key>.png
      * @param  editorSpecific Optional boolean that says if the resource is an editor specific resource, default is false.
      */
-    add(asset : Asset, filePath? : string, editorSpecific? : boolean) {
-        filePath = filePath || this._store.getState().project.home + "/resources/" + asset.key + "." + this._fileExtensionFromType(asset.type);
-        if (editorSpecific) {
-            asset.key = EDITOR_SPECIFIC_IMAGE_PREFIX + asset.key;
+    add(assets : LoadingAsset[]) {
+        if (loader.loading) {
```
Comment (jeffswenson) [inline src/duckling/project/asset.service.ts:51]: You need a return here.

File: src/duckling/project/asset.service.ts [hunk 6]
```diff
@@ -40,39 +47,46 @@ export class AssetService {
...
-            if (asset.type === "FontTTF") {
-                this._loadFont(asset, filePath, editorSpecific)
-            } else {
-                // resource-loader library can only load one thing at a time so if 
-                // we are already loaded cache off the asset to load once it's done
-                if (!loader.loading) {
-                    loader.add(asset.key, filePath).load();
-                    loader.once('complete', () => this._onLoadedComplete(asset, editorSpecific));
+        
+        for (let assetToLoad of assets) {
```
Comment (jeffswenson) [inline src/duckling/project/asset.service.ts]: I think you could modularize this function a little better.

**Review by jeffswenson — CHANGES_REQUESTED** [review]

File: src/duckling/project/asset.service.ts (hunk 6 above)
Comment (tlein) [reply src/duckling/project/asset.service.ts]: fixed

File: src/duckling/project/asset.service.ts (hunk 4 above)
Comment (tlein) [reply src/duckling/project/asset.service.ts]: fixed

File: src/duckling/project/asset.service.ts (hunk 5 above)
Comment (tlein) [reply src/duckling/project/asset.service.ts:51]: fixed

File: src/duckling/project/asset.service.ts (hunk 3 above)
Comment (tlein) [reply src/duckling/project/asset.service.ts]: fixed

**Pull request merged**